CCBluetooth/Classes/BluetoothDateTime.swift

```diff
@@ -0,0 +1,112 @@
+        } else if (daylightOffset == 0) {
+            dstOffsetValue = dst.DSTStandardTime.rawValue
+        }
+        
+        return dstOffsetValue
+    }
+
+    public func dateFromData(data: NSData) -> String {
+        var dateComponents = DateComponents()
+        
+        let year = (data.subdata(with: yearRange) as NSData)
+        let swappedYear = year.swapUInt16Data()
+        dateComponents.year = Int(strtoul(swappedYear.toHexString(), nil, 16))
+        
+        let month = (data.subdata(with: monthRange) as NSData)
+        dateComponents.month = Int(strtoul(month.toHexString(), nil, 16))
+        
+        let day = (data.subdata(with: dayRange) as NSData)
+        dateComponents.day = Int(strtoul(day.toHexString(), nil, 16))
+        
+        let hours = (data.subdata(with: hoursRange) as NSData)
+        dateComponents.hour = Int(strtoul(hours.toHexString(), nil, 16))
+        
+        let minutes = (data.subdata(with: minutesRange) as NSData)
+        dateComponents.minute = Int(strtoul(minutes.toHexString(), nil, 16))
+        
+        let seconds = (data.subdata(with: secondsRange) as NSData)
+        dateComponents.second = Int(strtoul(seconds.toHexString(), nil, 16))
+        
+        let measurementDate = NSCalendar.current.date(from: dateComponents as DateComponents)
+        let timeZoneData: NSData = (data.subdata(with: timeZoneByteRange) as NSData)
+        var timeZoneInt: NSInteger = 0
+        timeZoneData.getBytes(&timeZoneInt, length: 1)
+        timeZoneInt = (((timeZoneInt - 256) / timeZoneStepSizeMin60) * secondsInHour)
+        let tz: TimeZone = NSTimeZone(forSecondsFromGMT: timeZoneInt) as TimeZone
+        
+        var dstOffsetInSeconds: TimeInterval!
+        let dstOffsetCode = (data.subdata(with: dstOffsetByteRange) as NSData!)
+        var dstOffset: NSInteger = 0
+        dstOffsetCode?.getBytes(&dstOffset, length: 1)
+        
+        if ((dstOffset == dst.DSTStandardTime.rawValue) && (tz.daylightSavingTimeOffset() == 0)) {
+            dstOffsetInSeconds = tz.daylightSavingTimeOffset()
+        } else if ((dstOffset == dst.DSTPlusHourHalf.rawValue) && (Int(tz.daylightSavingTimeOffset()) != (secondsInHour / 2))) {
+            dstOffsetInSeconds = TimeInterval(Int(tz.daylightSavingTimeOffset()) - (secondsInHour / 2))
+        } else if ((dstOffset == dst.DSTPlusHourOne.rawValue) && (Int(tz.daylightSavingTimeOffset()) != secondsInHour)) {
+            dstOffsetInSeconds = TimeInterval(Int(tz.daylightSavingTimeOffset()) - secondsInHour);
+        } else if ((dstOffset == dst.DSTPlusHoursTwo.rawValue) && (Int(tz.daylightSavingTimeOffset()) != (secondsInHour * 2))) {
+            dstOffsetInSeconds = TimeInterval(Int(tz.daylightSavingTimeOffset()) - (secondsInHour * 2));
+        }
+        
+        dateComponents.second = dateComponents.second! - Int(dstOffsetInSeconds)
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss ZZZZ"
```

Comment:
Should these date conversion methods be move to CCToolbox? Just wondering. If they are specific to Bluetooth then they might make sense to leave them here. But if they have applications elsewhere, maybe they should be moved to the CCToolbox so that apps don't need to pull in CCBluetooth in order to get the date conversion methods. Just a thought (no need to move it if you don't think it makes sense).